os-eap-legacy/os-eap-launch/added/launch/datasource-common.sh

```diff
@@ -75,6 +75,8 @@ function inject_datasources_common() {
 
   tx_datasource="$(inject_tx_datasource)"
   if [ -n "$tx_datasource" ]; then
+    # KIECLOUD-598 We need to escape the & if it isn't already escaped
+    tx_datasource=$(echo "$tx_datasource" | sed 's|\([^\\]\)&amp|\1\\\&amp|g')
```

Comment:
is there a way to not touch the legacy files? 
Is just the EJB_TIMER datasource that is not working with url parameters, right?

these legacy ones are copied from EAP modules, if we need to add these changes on the eap scripts these needs also to be added on the jboss-eap-modules in order to avoid regressions when we start using the jboss-eap-modules/datasources instead the legacy one.

Reply:
Yes @spolti I agree with you, we don't need to touch the legacy files and so I'll revert these and all the changes I made in the "exploratory phase" to help me in understanding which datasources could have been impacted by this issue.

Reply:
I think that I found the right solution to avoid any change to the legacy modules we copied from the EAP modules, what do you think?